src/darwin/Framework/CHIP/CHIPDevice.h

```diff
@@ -49,6 +72,91 @@ NS_ASSUME_NONNULL_BEGIN
                 maxInterval:(uint16_t)maxInterval
               reportHandler:(void (^)(NSArray * _Nullable value, NSError * _Nullable error))reportHandler
     subscriptionEstablished:(nullable void (^)(void))subscriptionEstablishedHandler;
+
+/**
+ * Read attribute in a designated attribute path
+ *
+ * @param completion  response handler will receive either value or error. value will be an NSArray object with NSDictionary
+ * elements. Each NSDictionary will have "endpointId", "clusterId", "attributeId", "status" and "data" keys. "endpointId",
+ * "clusterId", "attributeId" and "status" will be mapped to NSNumber objects. "status" with 0 value indicates success and non-zero
+ * value indicates failure. "data" key is present only when "status" value is 0. "data" key will be mapped to an NSDictionary
+ * object, representing attribute value of the path. NSDictionary representing attribute value will contain "type" and "value" keys.
+ *                        "type" will be mapped to "SignedInteger", "UnsignedInteger", "UTF8String", "OctetString", "Float",
+ * "Double", "Boolean", "Null", "Structure", "Array" or "List. "value" will be mapped to an NSNumber, NSString, nil or NSArray
```

Comment:
I meant that element[@"value"] is nil (i.e., no key in the dictionary) in case element[@"type"] is @"Null". But I agree that the comment needs more clarification.